sql: populate database_name when selecting from crdb_internal.create_statements

RichardJCai opened the pull request:
Previously, SELECTING from crdb_internal.create_statements and not
using the virtual index would result in database_name being nil.
This was because we did not pass the db descriptor into populate.

Release justification: Low risk, bug fix, internal table
Release note (sql change): selecting from crdb_internal.create_statements
will now correctly populate the database_name when the virtual index is
not used. Internal table.

Fixes #59561

RichardJCai committed:
sql: populate database_name when selecting from crdb_internal.create_statements

Previously, SELECTING from crdb_internal.create_statements and not
using the virtual index would result in database_name being nil.
This was because we did not pass the db descriptor into populate.

Release justification: Low risk, bug fix, internal table
Release note (bug fix): selecting from crdb_internal.create_statements
will now correctly populate the database_name when the virtual index is
not used. Internal table.

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/61201)
<!-- Reviewable:end -->

rafiss reviewed (approved):
awesome! nice fix

looks like a test needs updating -- i don't quite understand what's wrong with it now

RichardJCai commented:
> awesome! nice fix
> 
> looks like a test needs updating -- i don't quite understand what's wrong with it now

Updated test, it was failing because it was specifying database in the where clause but the database previously was being incorrectly returned as NULL for vtables.

Also removed virtualOnce completely since only `crdb_internal.create_statements` was using it.

RichardJCai commented:
Thanks for the review!
bors r=rafiss

RichardJCai commented:
Looks like a `create_statements` test is flaking cause of this

bors r-

RichardJCai commented:
Nvm checked the diff, looks like I need to rebase

RichardJCai commented:
bors r=rafiss

RichardJCai commented:
For some reason the license/cla is hanging? Gonna retry bors

bors retry

rafiss commented:
@RichardJCai here's a shell alias i have set up to re-trigger the cla check:

```
function cla() {
  curl -H "Authorization: token $(git config --get prmaster.githubToken)" \
    -d '{"state": "success", "context":"license/cla", "description": "curl"}' \
    https://api.github.com/repos/cockroachdb/cockroach/statuses/$1
}
```
usage:
```
$ cla <commit sha>
```

RichardJCai commented:
bors r=rafiss

RichardJCai commented:
> @RichardJCai here's a shell alias i have set up to re-trigger the cla check:
> 
> ```
> function cla() {
>   curl -H "Authorization: token $(git config --get prmaster.githubToken)" \
>     -d '{"state": "success", "context":"license/cla", "description": "curl"}' \
>     https://api.github.com/repos/cockroachdb/cockroach/statuses/$1
> }
> ```
> 
> usage:
> 
> ```
> $ cla <commit sha>
> ```

This worked, thanks!
